src/components/binding/binding.js

```diff
@@ -140,25 +149,72 @@ genotet.BindingView = function(viewName, params) {
     .on('genotet.addTrack', function() {
       var track = this.data.tracks.slice(-1).pop();
       this.loader.loadFullTrack(this.data.tracks.length, track.fileName,
-        this.data.chr);
+        this.data.chr, true);
     }.bind(this))
     .on('genotet.removeTrack', function(event, trackIndex) {
       this.data.tracks.splice(trackIndex, 1);
       this.renderer.render();
       this.panel.updateTracks();
     }.bind(this))
-    .on('genotet.gene', function(event, data) {
+    .on('genotet.updateTrack', function(event, data) {
       this.data.tracks[data.trackIndex].fileName = data.fileName;
       this.loader.loadFullTrack(data.trackIndex, data.fileName,
-        this.data.chr);
+        this.data.chr, false);
+    }.bind(this))
+    .on('genotet.loadBindingList', function() {
+      this.loader.loadBindingList();
     }.bind(this));
 
   $(this.loader)
     .on('genotet.chr', function(event, chr) {
       this.panel.updateChr(chr);
     }.bind(this))
-    .on('genotet.track', function() {
+    .on('genotet.addPanelTrack', function() {
       this.panel.updateTracks();
+    }.bind(this))
+    .on('genotet.updateTrackList', function() {
+      this.panel.updateTrackList();
+    }.bind(this));
+
+  // Set up link callbacks.
+  $(this)
+    .on('genotet.updateTrack', function(event, data) {
+      var sourceGene = data.id.split(',')[0];
```

Comment:
You need to describe in comments for accessing array index. You can also type annotate data.